src/mixer.rs

```diff
@@ -0,0 +1,428 @@
+//! Mixer API - Simple Mixer API for mixer control
+//!
+use std::ffi::CString;
+use std::{ptr, mem};
+
+use alsa;
+use super::card;
+use super::error::*;
+
+const SELEM_ID_SIZE: usize = 64;
+
+/// wraps [snd_mixer_t](http://www.alsa-project.org/alsa-doc/alsa-lib/group___mixer.html)
+pub struct Mixer(*mut alsa::snd_mixer_t);
+
+impl Mixer {
+    /// Creates a new iterator for a specific card using the cards index and
+    /// using hw name, i.e. `hw:0`
+    pub fn new(c: &card::Card) -> Result<Mixer> {
```

Comment:
Better to provide string as input; ioplugs can have mixers and they do not have the syntax hw:<number>